src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx

```diff
@@ -136,12 +139,14 @@ export const DataSourceSelectable = ({
         setDataSourceOptionList(
           consolidateDataSourceGroups(
             dataSetResults as DataSetWithDataSource[],
-            dataSources.filter((ds) => !ds.getMetadata().ui.selector.displayDatasetsAsSource)
+            newHomePageEnabled && !queryEnhancementsEnabled
+              ? []
+              : dataSources.filter((ds) => !ds.getMetadata().ui.selector.displayDatasetsAsSource)
```

Comment:
what does this do? title says s3 but logic doesn't seem s3 specific

Reply:
I should change the title to make it more clear.